npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -97,13 +97,11 @@ export async function makeWebpackConfig (
   if (!userWebpackConfig && !frameworkWebpackConfig) {
     debug('Not user or framework webpack config received. Trying to automatically source it')
 
-    const { default: findUp } = await importModule('find-up')
-
     configFile = await findUp(configFiles, { cwd: projectRoot } as { cwd: string })
 
     if (configFile) {
       debug('found webpack config %s', configFile)
-      const sourcedConfig = await importModule(configFile)
+      const sourcedConfig = configFile.endsWith('.mjs') ? await import(configFile) : require(configFile)
```

Comment:
Yeah I was reverting to what was there before, but I confirmed it doesn't work. Turns out the TS compilation will convert `import` to a require which is not what we want (see [here](https://github.com/TypeStrong/ts-node/discussions/1290)). There are some workarounds, I'll check both webpack and vite since we do similar things here

Reply:
Going over this PR now -- we can sync on this if needed, I've been dealing with module stuff lately. Not saying I have a perfect solution, but we can explore some things.

If we change the `tsconfig` to `"module": "esnext" it will **not** generate `require`, but maintain the original `import`.

Actually, thinking about it, we have the **exact same problem** in `packages/server`, where we need to grab their `cypress.config`, which can be:

- `"type": "module"` with `js`
- `"type": "CommonJS"` (the default) with `.mjs`
- `.ts` with "module": "CommonJS" in `tsconfig.json`
- `.ts` with "module": "esnext" in `tsconfig.json`

It's also possible to have `.cjs` and, believe it or not, `.mjs` and `.mts`. 

I think we probably want to have one single, uniform way to handle "all the things" if possible. The only difference, I guess, would be that the binary ships `ts-node`, so we know we can use that, but `npm/webpack-dev-server` cannot rely on `ts-node` been installed.